pkg/app/pipedv1/controller/scheduler.go

```diff
@@ -603,14 +603,25 @@ func (s *scheduler) executeStage(sig StopSignal, ps *model.PipelineStage) (final
 			TargetDeploymentSource:  tds.ToPluginDeploySource(),
 		},
 	})
-	// do not return error if the context is already canceled.
-	// this occurs when the stage is canceled.
-	// otherwise, return the error.
+
+	// Return stage failure status if the plugin execution failed and the context is not cancelled.
+	// The cause probably is the plugin binary is not available.
 	if err != nil && ctx.Err() == nil {
 		s.logger.Error("failed to execute stage", zap.String("stage-name", ps.Name), zap.Error(err))
 		return model.StageStatus_STAGE_FAILURE
 	}
 
+	// Handle case where res is nil.
```

Comment:
We should handle the case `err != nil && ctx.Err() == nil`, which is not handled in the current implementation, instead of the case `res == nil` because this is gRPC invocation, its response will nil only when its error is not nil. WDYT?

Reply:
@Warashi thanks 👍 I address your point by https://github.com/pipe-cd/pipecd/pull/6275/commits/95c78a441067a2556e8a1e68cc3c8a2426b3d17f